### PR TITLE
Add function onHover DataRow in DataTable

### PR DESCRIPTION
**Pull request by QunNguyen**
Add function onHover DataRow in DataTable listen event onHover Flutter web. This is because DataRow not listen event when user hover row.

https://github.com/flutter/flutter/assets/92642706/2ad71744-2d3b-4137-a16b-b0c08c90d9d3

*List which issues are fixed by this PR. You must list at least one issue.
Fixes b128659

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Review by Piinks — COMMENTED** [review]
Hi @QunNguyen welcome! Thank you for contributing!
Can you add a test that verifies that this works as expected?

**Comment by QunNguyen** [comment]
Hi @Piinks.
What test do I need to take? Can you guide me ?

**Comment by Piinks** [comment]
@QunNguyen this page may help you with testing: https://github.com/flutter/flutter/blob/master/docs/contributing/testing/Writing-Effective-Tests.md

**Comment by QunNguyen** [comment]
Hi @Piinks, 
I have update code test with my code.
Screenshot 2024-07-12 110949
Can you check for me? 
Thank you.

File: packages/flutter/lib/src/material/data_table.dart [hunk 1]
```diff
@@ -171,6 +173,11 @@ class DataRow {
   /// that particular cell.
   final ValueChanged<bool?>? onSelectChanged;
 
+
+  /// onHover returns true or false when the row is hovered
+  /// onHover works independently of onSelectChanged
```
Comment (Piinks) [inline packages/flutter/lib/src/material/data_table.dart:178]: Can you rewrite this doc looking at `onSelectChanged` above as an example?

File: packages/flutter/test/material/data_table_test.dart [hunk 2]
```diff
@@ -46,6 +46,9 @@ void main() {
             onSelectChanged: (bool? selected) {
               log.add('row-selected: ${dessert.name}');
             },
+            onHover: (bool? selected) {
```
Comment (Piinks) [inline packages/flutter/test/material/data_table_test.dart:49]: This test is not quite sufficient. This sets the property, but it does not verify that the callback is triggered by a hover event.

**Comment by Piinks** [comment]
Hey @QunNguyen, would you like to continue working on this change?

**Comment by Piinks** [comment]
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to submit a new PR if you have the time to address the review comments. Thanks!

**Pull request closed without merging**